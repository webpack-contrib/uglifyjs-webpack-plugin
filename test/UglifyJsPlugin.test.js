import UglifyJsPlugin from '../src/index';

describe('UglifyJsPlugin', () => {
  it('export as function', () => {
    expect(typeof new UglifyJsPlugin().apply).toBe('function');
  });

  it('isSourceMap method', () => {
    const rawSourceMap = {
      version: 3,
      file: 'min.js',
      names: ['bar', 'baz', 'n'],
      sources: ['one.js', 'two.js'],
      sourceRoot: 'http://example.com/www/js/',
      mappings: 'CAAC,IAAI,IAAM,SAAUA,GAClB,OAAOC,IAAID;CCDb,IAAI,IAAM,SAAUE,GAClB,OAAOA',
    };
    const emptyRawSourceMap = {
      version: 3,
      sources: [],
      mappings: '',
    };

    expect(UglifyJsPlugin.isSourceMap(null)).toBe(false);
    expect(UglifyJsPlugin.isSourceMap()).toBe(false);
    expect(UglifyJsPlugin.isSourceMap({})).toBe(false);
    expect(UglifyJsPlugin.isSourceMap([])).toBe(false);
    expect(UglifyJsPlugin.isSourceMap('foo')).toBe(false);
    expect(UglifyJsPlugin.isSourceMap({ version: 3 })).toBe(false);
    expect(UglifyJsPlugin.isSourceMap({ sources: '' })).toBe(false);
    expect(UglifyJsPlugin.isSourceMap({ mappings: [] })).toBe(false);
    expect(UglifyJsPlugin.isSourceMap({ version: 3, sources: '' })).toBe(false);
    expect(UglifyJsPlugin.isSourceMap({ version: 3, mappings: [] })).toBe(false);
    expect(UglifyJsPlugin.isSourceMap({ sources: '', mappings: [] })).toBe(false);
    expect(UglifyJsPlugin.isSourceMap({ version: 3, sources: '', mappings: [] })).toBe(false);
    expect(UglifyJsPlugin.isSourceMap(rawSourceMap)).toBe(true);
    expect(UglifyJsPlugin.isSourceMap(emptyRawSourceMap)).toBe(true);
  });
});
