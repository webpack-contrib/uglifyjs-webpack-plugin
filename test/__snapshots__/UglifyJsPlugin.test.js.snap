// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UglifyJsPlugin buildError method 1`] = `
[Error: test.js from UglifyJs
Message]
`;

exports[`UglifyJsPlugin buildError method 2`] = `
[Error: test.js from UglifyJs
Message [test.js:1,1]]
`;

exports[`UglifyJsPlugin buildError method 3`] = `
[Error: test.js from UglifyJs
Message [./one.js:1,1][test.js:1,1]]
`;

exports[`UglifyJsPlugin buildError method 4`] = `
[Error: test.js from UglifyJs
Stack]
`;

exports[`UglifyJsPlugin buildSourceMap method 1`] = `
SourceMapConsumer {
  "_absoluteSources": Array [
    "http://example.com/www/js/one.js",
    "http://example.com/www/js/two.js",
  ],
  "_mappings": "CAAC,IAAI,IAAM,SAAUA,GAClB,OAAOC,IAAID;CCDb,IAAI,IAAM,SAAUE,GAClB,OAAOA",
  "_names": ArraySet {
    "_array": Array [
      "bar",
      "baz",
      "n",
    ],
    "_set": Map {
      "bar" => 0,
      "baz" => 1,
      "n" => 2,
    },
  },
  "_sourceMapURL": undefined,
  "_sources": ArraySet {
    "_array": Array [
      "one.js",
      "two.js",
    ],
    "_set": Map {
      "one.js" => 0,
      "two.js" => 1,
    },
  },
  "file": "test.js",
  "sourceRoot": "http://example.com/www/js/",
  "sourcesContent": null,
}
`;

exports[`UglifyJsPlugin buildWarning method 1`] = `"UglifyJs Plugin: Warning [test.js:1,1]"`;

exports[`UglifyJsPlugin buildWarning method 2`] = `"UglifyJs Plugin: Warning [test.js:1,1]"`;

exports[`UglifyJsPlugin buildWarning method 3`] = `"UglifyJs Plugin: Warning [test.js:1,1]"`;

exports[`UglifyJsPlugin buildWarning method 4`] = `"UglifyJs Plugin: Warning [./one.js:1,1]"`;

exports[`UglifyJsPlugin buildWarning method 5`] = `"UglifyJs Plugin: Warning [./one.js:1,1]"`;

exports[`UglifyJsPlugin buildWarning method 6`] = `null`;

exports[`UglifyJsPlugin should regenerate hash: 4.4.b3a3e1129148263bd259.js 1`] = `"(window.webpackJsonp=window.webpackJsonp||[]).push([[4],{5:function(n,p){n.exports=\\"async-dep\\"}}]);"`;

exports[`UglifyJsPlugin should regenerate hash: AsyncImportExport.80ec742362058f8c35a8.js 1`] = `"!function(c){function e(e){for(var t,n,r=e[0],o=e[1],u=0,i=[];u<r.length;u++)n=r[u],a[n]&&i.push(a[n][0]),a[n]=0;for(t in o)Object.prototype.hasOwnProperty.call(o,t)&&(c[t]=o[t]);for(f&&f(e);i.length;)i.shift()()}var n={},a={0:0};function l(e){if(n[e])return n[e].exports;var t=n[e]={i:e,l:!1,exports:{}};return c[e].call(t.exports,t,t.exports,l),t.l=!0,t.exports}l.e=function(u){var e,t=[],n=a[u];if(0!==n)if(n)t.push(n[2]);else{var r=new Promise(function(e,t){n=a[u]=[e,t]});t.push(n[2]=r);var o,i=document.createElement(\\"script\\");i.charset=\\"utf-8\\",i.timeout=120,l.nc&&i.setAttribute(\\"nonce\\",l.nc),i.src=l.p+\\"\\"+(e=u)+\\".\\"+({}[e]||e)+\\".\\"+{4:\\"b3a3e1129148263bd259\\"}[e]+\\".js\\",o=function(e){i.onerror=i.onload=null,clearTimeout(c);var t=a[u];if(0!==t){if(t){var n=e&&(\\"load\\"===e.type?\\"missing\\":e.type),r=e&&e.target&&e.target.src,o=new Error(\\"Loading chunk \\"+u+\\" failed.\\\\n(\\"+n+\\": \\"+r+\\")\\");o.type=n,o.request=r,t[1](o)}a[u]=void 0}};var c=setTimeout(function(){o({type:\\"timeout\\",target:i})},12e4);i.onerror=i.onload=o,document.head.appendChild(i)}return Promise.all(t)},l.m=c,l.c=n,l.d=function(e,t,n){l.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:n})},l.r=function(e){\\"undefined\\"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:\\"Module\\"}),Object.defineProperty(e,\\"__esModule\\",{value:!0})},l.t=function(t,e){if(1&e&&(t=l(t)),8&e)return t;if(4&e&&\\"object\\"==typeof t&&t&&t.__esModule)return t;var n=Object.create(null);if(l.r(n),Object.defineProperty(n,\\"default\\",{enumerable:!0,value:t}),2&e&&\\"string\\"!=typeof t)for(var r in t)l.d(n,r,function(e){return t[e]}.bind(null,r));return n},l.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return l.d(t,\\"a\\",t),t},l.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},l.p=\\"\\",l.oe=function(e){throw console.error(e),e};var t=window.webpackJsonp=window.webpackJsonp||[],r=t.push.bind(t);t.push=e,t=t.slice();for(var o=0;o<t.length;o++)e(t[o]);var f=r;l(l.s=4)}({4:function(e,t,n){n.e(4).then(function(){n(5),console.log(\\"Good\\")}.bind(null,n)).catch(n.oe),e.exports=\\"Awesome\\"}});"`;

exports[`UglifyJsPlugin should regenerate hash: errors 1`] = `Array []`;

exports[`UglifyJsPlugin should regenerate hash: importExport.b3fe6c6dcbdc695a4974.js 1`] = `"!function(t){var n={};function o(e){if(n[e])return n[e].exports;var r=n[e]={i:e,l:!1,exports:{}};return t[e].call(r.exports,r,r.exports,o),r.l=!0,r.exports}o.m=t,o.c=n,o.d=function(e,r,t){o.o(e,r)||Object.defineProperty(e,r,{enumerable:!0,get:t})},o.r=function(e){\\"undefined\\"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:\\"Module\\"}),Object.defineProperty(e,\\"__esModule\\",{value:!0})},o.t=function(r,e){if(1&e&&(r=o(r)),8&e)return r;if(4&e&&\\"object\\"==typeof r&&r&&r.__esModule)return r;var t=Object.create(null);if(o.r(t),Object.defineProperty(t,\\"default\\",{enumerable:!0,value:r}),2&e&&\\"string\\"!=typeof r)for(var n in r)o.d(t,n,function(e){return r[e]}.bind(null,n));return t},o.n=function(e){var r=e&&e.__esModule?function(){return e.default}:function(){return e};return o.d(r,\\"a\\",r),r},o.o=function(e,r){return Object.prototype.hasOwnProperty.call(e,r)},o.p=\\"\\",o(o.s=2)}([,,function(e,r,t){var n=t(3);e.exports=function(){var e=n,r=\\"baz\\"+Math.random();return function(){return{a:e+n.bar+r,b:e,baz:r}}}},function(e,r){e.exports=\\"foo\\",e.exports.bar=\\"bar\\"}]);"`;

exports[`UglifyJsPlugin should regenerate hash: js.10e5726eb117daad9196.js 1`] = `"!function(n){var r={};function o(e){if(r[e])return r[e].exports;var t=r[e]={i:e,l:!1,exports:{}};return n[e].call(t.exports,t,t.exports,o),t.l=!0,t.exports}o.m=n,o.c=r,o.d=function(e,t,n){o.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:n})},o.r=function(e){\\"undefined\\"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:\\"Module\\"}),Object.defineProperty(e,\\"__esModule\\",{value:!0})},o.t=function(t,e){if(1&e&&(t=o(t)),8&e)return t;if(4&e&&\\"object\\"==typeof t&&t&&t.__esModule)return t;var n=Object.create(null);if(o.r(n),Object.defineProperty(n,\\"default\\",{enumerable:!0,value:t}),2&e&&\\"string\\"!=typeof t)for(var r in t)o.d(n,r,function(e){return t[e]}.bind(null,r));return n},o.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return o.d(t,\\"a\\",t),t},o.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},o.p=\\"\\",o(o.s=0)}([function(e,t){e.exports=function(){console.log(7)}}]);"`;

exports[`UglifyJsPlugin should regenerate hash: mjs.0c0df0657cc9467f6264.js 1`] = `"!function(r){var n={};function o(e){if(n[e])return n[e].exports;var t=n[e]={i:e,l:!1,exports:{}};return r[e].call(t.exports,t,t.exports,o),t.l=!0,t.exports}o.m=r,o.c=n,o.d=function(e,t,r){o.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:r})},o.r=function(e){\\"undefined\\"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:\\"Module\\"}),Object.defineProperty(e,\\"__esModule\\",{value:!0})},o.t=function(t,e){if(1&e&&(t=o(t)),8&e)return t;if(4&e&&\\"object\\"==typeof t&&t&&t.__esModule)return t;var r=Object.create(null);if(o.r(r),Object.defineProperty(r,\\"default\\",{enumerable:!0,value:t}),2&e&&\\"string\\"!=typeof t)for(var n in t)o.d(r,n,function(e){return t[e]}.bind(null,n));return r},o.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return o.d(t,\\"a\\",t),t},o.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},o.p=\\"\\",o(o.s=1)}([,function(e,t,r){\\"use strict\\";r.r(t);module.exports=function(){console.log(7)}}]);"`;

exports[`UglifyJsPlugin should regenerate hash: warnings 1`] = `Array []`;

exports[`UglifyJsPlugin should works (without options): errors 1`] = `Array []`;

exports[`UglifyJsPlugin should works (without options): main.c0c3b643694d2bccfa85.js 1`] = `"!function(n){var r={};function o(e){if(r[e])return r[e].exports;var t=r[e]={i:e,l:!1,exports:{}};return n[e].call(t.exports,t,t.exports,o),t.l=!0,t.exports}o.m=n,o.c=r,o.d=function(e,t,n){o.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:n})},o.r=function(e){\\"undefined\\"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:\\"Module\\"}),Object.defineProperty(e,\\"__esModule\\",{value:!0})},o.t=function(t,e){if(1&e&&(t=o(t)),8&e)return t;if(4&e&&\\"object\\"==typeof t&&t&&t.__esModule)return t;var n=Object.create(null);if(o.r(n),Object.defineProperty(n,\\"default\\",{enumerable:!0,value:t}),2&e&&\\"string\\"!=typeof t)for(var r in t)o.d(n,r,function(e){return t[e]}.bind(null,r));return n},o.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return o.d(t,\\"a\\",t),t},o.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},o.p=\\"\\",o(o.s=0)}([function(e,t){e.exports=function(){console.log(7)}}]);"`;

exports[`UglifyJsPlugin should works (without options): warnings 1`] = `Array []`;
